compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -67,9 +111,20 @@ bool areLinalgOpsFusableUsingTileAndFuse(OpOperand &use) {
   // 2. Consumer is elementwise parallel.
   if (consumer.getNumLoops() != consumer.getNumParallelLoops()) return false;
 
-  // 3. In consumer the result of producer is accessed using identity indexing.
+  // 3. Check if a reduction result is used in the following elementwise
+  // operation with broadcast. If so, we can fuse the reduction into the
+  // elementwise op. The elementwise op on the reduced dimension will be
+  // serialized to match the workgroup counts of the fused operations.
+  // Otherwise, check if the result of producer is accessed using identity
+  // indexing.
   AffineMap consumerIndexingMap = consumer.getTiedIndexingMap(&use);
-  if (!consumerIndexingMap.isIdentity()) return false;
+  if (clFuseReductionWithElementwise && isSimpleReduction(producer) &&
+      isBroadcast(consumerIndexingMap) && isElementwise(consumer) &&
+      producer->getOperand(0).getType() == consumer->getResult(0).getType()) {
```

Comment:
Is this condition enough to ensure that the reduction loop and the broadcasted dimension match? What if the dimension is dynamic?

Reply:
Updated the code to check if the same indexing map is used for reduction and broadcast, which guarantees that the broadcast recovers the original shape of the input to the reduction.